passage/src/main/java/id/passage/android/PassagePasskey.kt

```diff
@@ -0,0 +1,94 @@
+package id.passage.android
+
+import android.app.Activity
+import id.passage.android.api.LoginAPI
+import id.passage.android.api.RegisterAPI
+import id.passage.android.exceptions.LoginWithPasskeyException
+import id.passage.android.exceptions.RegisterWithPasskeyException
+import id.passage.android.model.AuthResult
+import id.passage.android.model.AuthenticatorAttachment
+import id.passage.android.model.LoginWebAuthnFinishRequest
+import id.passage.android.model.LoginWebAuthnStartRequest
+import id.passage.android.model.RegisterWebAuthnFinishRequest
+import id.passage.android.model.RegisterWebAuthnStartRequest
+import okhttp3.OkHttpClient
+
+class PassagePasskey(
+    private val passageClient: OkHttpClient,
+    private val activity: Activity,
+) {
+    /**
+     * Register user with passkey
+     *
+     * Create a user, prompt the user to create a passkey, and register the user.
+     * @param identifier valid email or E164 phone number
+     * @param options optional configuration for passkey creation
+     * @return PassageAuthResult
+     * @throws RegisterWithPasskeyException
+     */
+    suspend fun register(
+        identifier: String,
+        options: PasskeyCreationOptions? = null,
+    ): AuthResult {
+        try {
+            val registerAPI = RegisterAPI(Passage.BASE_PATH, passageClient)
+            // Get Create Credential challenge from Passage
+            val authenticatorAttachment =
+                options?.authenticatorAttachment
+                    ?: AuthenticatorAttachment.platform
+            val webauthnStartRequest = RegisterWebAuthnStartRequest(identifier, authenticatorAttachment)
+            val webauthnStartResponse = registerAPI.registerWebauthnStart(Passage.appId, webauthnStartRequest)
+            // Use Create Credential challenge to prompt user to create a passkey
+            val createCredOptionsJson = PasskeyUtils.getCreateCredentialOptionsJson(webauthnStartResponse.handshake)
+            val createCredResponse = PasskeyUtils.createPasskey(createCredOptionsJson, activity)
+            // Complete registration and authenticate the user
+            val handshakeResponse = PasskeyUtils.getCreateCredentialHandshakeResponse(createCredResponse)
+            val webauthnFinishRequest =
+                RegisterWebAuthnFinishRequest(
+                    handshakeId = webauthnStartResponse.handshake.id,
+                    handshakeResponse = handshakeResponse,
+                    userId = webauthnStartResponse.user?.id ?: "",
+                )
+            val authResponse = registerAPI.registerWebauthnFinish(Passage.appId, webauthnFinishRequest)
+            // Handle and return auth result
+            // handleAuthResult(authResponse.authResult)
```

Comment:
Shouldn't we be saving the tokens in the new way here?
```suggestion
            tokenStore.setTokens(authResponse.authResult)
```